feat(node): add identify behaviour

b-zee opened the pull request:
Using the identify behaviour to exchange node information and adding addresses to the routing table based on that.

b-zee committed:
feat(node): add identify behaviour

Using the identify behaviour to exchange node information and adding addresses to the routing table based on that.

joshuef commented on safenode/src/network/mod.rs:
```diff
@@ -203,12 +203,18 @@ impl SwarmDriver {
         let mdns = mdns::tokio::Behaviour::new(mdns_config, peer_id)?;
+
+        let identify_cfg =
+            libp2p::identify::Config::new("safe/0.1.0".to_string(), keypair.public())
```
can we pull this from a const? (is this semvar? I wonder how this might interact with the protocol folder and changes there... something for the huddle)

b-zee replied:
Good one, I think for now we should mostly ignore versions. As long as we don't go to major 1. But putting it in a const is probably a good idea.

b-zee committed:
refactor(node): add strings as const

b-zee committed:
feat(client): add passed in peer to routing table

Also fixes a problem where the client thinks it's connected to the
network, while we're not yet adding any node that we discover via mDNS.
We need to wait for the Identify behavior to kick in.

maqi commented on safenode/src/network/mod.rs:
```diff
@@ -280,6 +292,18 @@ impl Network {
         receiver.await?
     }
 
+    /// Dial the given peer at the given address.
```
the code block of the function is nothing related to `dial`?
though maybe the later on swarm will carry out `dial` before add to RT ?
